### PR TITLE
Adds support for multi-types invopop/jsonschema#134

### PR DESCRIPTION
**Pull request by PopescuStefanRadu**

### CODE REVIEW

File: schema.go [hunk 1]
```diff
@@ -92,3 +92,8 @@ var (
 // http://json-schema.org/latest/json-schema-validation.html#rfc.section.5.26
 // RFC draft-wright-json-schema-validation-00, section 5.26
 type Definitions map[string]*Schema
+
+// Type implements section 6.1.1
+type Type struct {
+	Types []string
+}
```
Comment (samlown) [inline schema.go:99]: Did you already consider this?

```suggestion
type Type []string
```

I'd also suggest adding helper methods to avoid all the messy assignments, instead of:
```
&Type{Types: []string{"string"}}
```
I'd expect:
```
MakeType("string")
```

With a method pattern that looks like:

```
func MakeType(typ ...string) Type
```

(I'm using `Make` in the method name as opposed to `New` reflecting the underlying slice instead of an object.)

File: reflect.go [hunk 2]
```diff
@@ -1112,6 +1125,33 @@ func (t *Schema) MarshalJSON() ([]byte, error) {
 	return append(b, m[1:]...), nil
 }
 
+// MarshalJSON implements json.Marshaler
+func (tp *Type) MarshalJSON() ([]byte, error) {
```
Comment (samlown) [inline reflect.go:1129]: I'd define these methods alongside the `Type` struct in the `schema.go` file.

File: reflect.go [hunk 3]
```diff
@@ -1112,6 +1125,33 @@ func (t *Schema) MarshalJSON() ([]byte, error) {
...
+	switch len(tp.Types) {
+	case 0:
+		return []byte("[]"), nil
```
Comment (samlown) [inline reflect.go:1132]: Is this correct? It should be tested either way.

File: reflect.go [hunk 4]
```diff
@@ -1112,6 +1125,33 @@ func (t *Schema) MarshalJSON() ([]byte, error) {
...
+// UnmarshalJSON implements json.Unm\arshaler
+func (tp *Type) UnmarshalJSON(data []byte) error {
+	err := json.Unmarshal(data, &tp.Types)
+	if err == nil {
+		return nil
+	}
+	var v string
+	err2 := json.Unmarshal(data, &v)
+	if err2 != nil {
+		return fmt.Errorf("could not read type into slice: %v, nor into string: %w", err.Error(), err2)
```
Comment (samlown) [inline reflect.go:1149]: I couldn't see any tests for this.

**Review by samlown — CHANGES_REQUESTED** [review]
Solid start! Thanks! I'd suggest refactoring the method naming and also cover more of the edge cases with tests.
